crates/interledger-service-util/src/balance_service.rs

```diff
@@ -105,30 +111,31 @@ where
                         triggered_by: &ilp_address,
                         data: &[],
                     }
-                        .build()
+                    .build()
                 })
                 .and_then(move |_| {
                     next.send_request(request)
                         .and_then(move |fulfill| {
-                            // TODO should we spawn a task to update the balances instead of doing it before returning the fulfill?
-                            // do we always return the fulfill, even if the state update fails? If so, I think we can.
-                            store.update_balances_for_fulfill(from.clone(), incoming_amount, to.clone(), outgoing_amount)
-                                .then(move |result| {
-                                    if result.is_err() {
-                                        error!("Error applying balance changes for fulfill from account: {} to account: {}. Incoming amount was: {}, outgoing amount was: {}", from.id(), to.id(), incoming_amount, outgoing_amount);
-                                    }
-                                    Ok(fulfill)
-                                })
+                            let fulfill_balance_update = store.update_balances_for_fulfill(
+                                from.clone(),
+                                incoming_amount,
+                                to.clone(),
+                                outgoing_amount,
+                            );
```

Comment:
It's important to log if this doesn't work because otherwise we'll have no idea it didn't

```suggestion
                            ).map_err(|_| error!("Error applying balance changes for fulfill from account: {} to account: {}. Incoming amount was: {}, outgoing amount was: {}", from.id(), to.id(), incoming_amount, outgoing_amount));
```